test/scenarios/outputs/complex_output_reference.yaml

```diff
@@ -0,0 +1,22 @@
+templateType: jsontemplate
+steps:
+  - name: complexOutput
+    template: |
+      {
+        "foo": "foo",
+        "bar": "bar"
+      }
+  - name: workflow
+    steps:
+      - name: foo
+        template:  |
+          { output: $.outputs.complexOutput.foo }
+      - name: bar
+        template: |
+          { output: $.outputs.complexOutput.bar }
+      - name: foobar
+        template: |
+          $.outputs.workflow.foo.output + " " + $.outputs.workflow.bar.output
```

Comment:
@sanpj2292 this is an example of referring child step output fields using dot notation

Reply:
Can the developer use `$.outputs.w1.c1.w2.c2` where w1= workflow-1, w2 = workflow-2, c1=simple-1, c2=simple-2 ? If no, do we have a validation in place for it ?

Reply:
we don't allow a workflow step inside another workflow step and there is validation for that so it will fail much before than this validation